snapcraft/parts/plugins/matter_plugin.py

```diff
@@ -0,0 +1,173 @@
+        part_info: infos.PartInfo,
+    ) -> None:
+        super().__init__(properties=properties, part_info=part_info)
+
+        self.matter_dir = part_info.part_build_dir
+        self.snap_arch = os.getenv("SNAP_ARCH")
+
+    @overrides
+    def get_build_packages(self) -> Set[str]:
+        return {
+            "wget",
+            "unzip",
+            "clang",
+            "pkg-config",
+            "git",
+            "cmake",
+            "ninja-build",
+            "libssl-dev",
+            "libdbus-1-dev",
+            "libglib2.0-dev",
+            "libavahi-client-dev",
+            "python3-venv",
+            "python3-dev",
+            "python3-pip",
+            "libgirepository1.0-dev",
+            "libcairo2-dev",
+            "libreadline-dev",
+            "generate-ninja",
+        }
+
+    @overrides
+    def get_build_environment(self) -> Dict[str, str]:
+        return {}
+
+    @overrides
+    def get_build_snaps(self) -> Set[str]:
+        return set()
+
+    @overrides
+    def get_build_commands(self) -> List[str]:
+        options = cast(MatterPluginProperties, self._options)
+        commands = []
+
+        if self.snap_arch == "arm64":
+            commands.extend(
+                [
+                    f"wget --no-verbose https://github.com/project-chip/zap/releases/download/"
+                    f"{options.matter_zap_version}/zap-linux-{self.snap_arch}.zip",
+                    f"unzip -o zap-linux-{self.snap_arch}.zip",
+                    "echo 'export ZAP_INSTALL_PATH=$PWD'",
+                ]
+            )
+
+        # Clone Matter repository if not present
+        commands.extend(
+            [
+                f"if [ ! -d matter ]; then git clone --depth 1 -b {options.matter_sdk_version} "
+                f"{MATTER_REPO} matter && cd matter; else cd matter || echo 'skip clone'; fi"
+            ]
+        )
+
+        # Checkout submodules for Linux platform
+        commands.extend(["scripts/checkout_submodules.py --shallow --platform linux"])
+
+        """ 
+        The project writes its data to /tmp which isn't persisted.
+
+        Setting TMPDIR env var when running the app isn't sufficient as 
+        chip_[config,counter,factory,kvs].ini still get written under /tmp.
+        The chip-tool currently has no way of overriding the default paths to
+        storage and security config files.
+
+        Snap does not allow bind mounting a persistent directory on /tmp, 
+        so we need to replace it in the source with another path, e.g. /mnt.
+        See the top-level layout definition which bind mounts a persisted
+        directory within the confined snap space on /mnt.
+        """
+        """Replace storage paths"""
+        commands.extend(
+            [
+                r"sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPLinuxStorage.h",
+                r"sed -i 's/\/tmp/\/mnt/g' src/platform/Linux/CHIPPlatformConfig.h",
+            ]
+        )
```

Comment:
I think you may want to put this into `/var/lib/matter` if you want to it persisted. But if it is default to `/tmp`, that should able to lose without issue.

Reply:
This has been placed into `/mnt`, and `/mnt` has been bound to `$SNAP_COMMON/mnt`, as shown in the example [here](https://github.com/MonicaisHer/chip-tool-snap/blob/test-matter-plugin/snap/snapcraft.yaml#L17-L19). Indeed, the location of persisted data could be improved to `/var/lib/matter` rather than `/mnt` on both the plugin side and consumer snap side.

Although it defaults to `/tmp`, the consumer snap needs the data to remain consistent after reboot.